metrics-core/src/test/java/com/codahale/metrics/Slf4jReporterTest.java

```diff
@@ -313,33 +310,54 @@ public void reportsTimerValues() {
         when(timer.getSnapshot()).thenReturn(snapshot);
         when(logger.isInfoEnabled(marker)).thenReturn(true);
 
-        infoReporter.report(map(),
+        infoReporter().report(map(),
                 map(),
                 map(),
                 map(),
                 map("test.another.timer", timer));
 
-        verify(logger).info(marker,
-                "type={}, name={}, count={}, min={}, max={}, mean={}, stddev={}, median={}, p75={}, p95={}, p98={}, p99={}, p999={}, mean_rate={}, m1={}, m5={}, m15={}, rate_unit={}, duration_unit={}",
-                "TIMER",
-                "prefix.test.another.timer",
-                1L,
-                300.0,
-                100.0,
-                200.0,
-                400.0,
-                500.0,
-                600.0,
-                700.0,
-                800.0,
-                900.0,
-                1000.0,
-                2.0,
-                3.0,
-                4.0,
-                5.0,
-                "events/second",
-                "milliseconds");
+        verifyInfo(expectedLog);
+    }
+
+    private Slf4jReporter infoReporter() {
+        return Slf4jReporter.forRegistry(registry)
+                .outputTo(logger)
+                .markWith(marker)
+                .prefixedWith("prefix")
+                .convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS)
+                .withLoggingLevel(Slf4jReporter.LoggingLevel.INFO)
+                .filter(MetricFilter.ALL)
+                .disabledMetricAttributes(disabledMetricAttributes)
+                .build();
+    }
+
+    private Slf4jReporter errorReporter() {
+        return Slf4jReporter.forRegistry(registry)
+                .outputTo(logger)
+                .markWith(marker)
+                .convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS)
+                .withLoggingLevel(Slf4jReporter.LoggingLevel.ERROR)
+                .filter(MetricFilter.ALL)
+                .disabledMetricAttributes(disabledMetricAttributes)
+                .build();
+    }
+
+    private void verifyError(String expectedLog) {
+        if (expectedLog == null) {
+            verify(logger, never()).error(marker, expectedLog);
+        } else {
+            verify(logger).error(marker, expectedLog);
+        }
+    }
+
+    private void verifyInfo(String expectedLog) {
+        if (expectedLog == null) {
+            verify(logger, never()).info(marker, expectedLog);
```

Comment:
It looks like this branch of the execution isn't used at all.

Reply:
the idea was to be consistent with how tests were written.
removed.